### PR TITLE
New NearestNeighbour1D algorithm

### PR DESCRIPTION
**Pull request by dbarbier**
Use dichotomy for nearest neighbour lookup of 1D points.

### CODE REVIEW

File: python/src/RegularGridNearestNeighbour_doc.i.in [hunk 1]
```diff
@@ -10,7 +10,7 @@ grid : :class:`~openturns.RegularGrid`
 
 See also
 --------
-NearestNeighbourAlgorithm, KDTree
+NearestNeighbourAlgorithm, NearestNeighbour1D, KDTree, NaiveNearestNeighbour
```
Comment (jschueller) [inline python/src/RegularGridNearestNeighbour_doc.i.in]: Must be added to the sphinx toctree too

Comment (jschueller) [reply python/src/RegularGridNearestNeighbour_doc.i.in]: and ChangeLog too

Comment (dbarbier) [reply python/src/RegularGridNearestNeighbour_doc.i.in]: Thanks, fixed

**Comment by regislebrun** [comment]
+1

**Pull request merged**